# Data Analysis on Video Game Sales

### Importing Libraries

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### Importing Data

In [3]:
path = "../Data/vgsales.csv"

In [4]:
df_games = pd.read_csv(path, index_col=0)

In [5]:
df_games.head()

Name Platform    Year         Genre Publisher  \
Rank                                                                      
1                   Wii Sports      Wii  2006.0        Sports  Nintendo   
2            Super Mario Bros.      NES  1985.0      Platform  Nintendo   
3               Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
4            Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
5     Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
1        41.49     29.02      3.77         8.46         82.74  
2        29.08      3.58      6.81         0.77         40.24  
3        15.85     12.88      3.79         3.31         35.82  
4        15.75     11.01      3.28         2.96         33.00  
5        11.27      8.89     10.22         1.00         31.37

### DataFrame summary

In [6]:
df_games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16598 entries, 1 to 16600
Data columns (total 10 columns):
Name            16598 non-null object
Platform        16598 non-null object
Year            16327 non-null float64
Genre           16598 non-null object
Publisher       16540 non-null object
NA_Sales        16598 non-null float64
EU_Sales        16598 non-null float64
JP_Sales        16598 non-null float64
Other_Sales     16598 non-null float64
Global_Sales    16598 non-null float64
dtypes: float64(6), object(4)
memory usage: 1.4+ MB


### About Features

- Year -> year values
- NA_Sales -> million dollars
- EU_Sales -> million dollars
- JP_Sales -> million dollars
- Other_Sales -> million dollars
- Global_Sales -> million dollars

### Removing Rows with NaN values

In [7]:
df_games = df_games.dropna(how='any',axis=0)

In [8]:
df_games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16291 entries, 1 to 16600
Data columns (total 10 columns):
Name            16291 non-null object
Platform        16291 non-null object
Year            16291 non-null float64
Genre           16291 non-null object
Publisher       16291 non-null object
NA_Sales        16291 non-null float64
EU_Sales        16291 non-null float64
JP_Sales        16291 non-null float64
Other_Sales     16291 non-null float64
Global_Sales    16291 non-null float64
dtypes: float64(6), object(4)
memory usage: 1.4+ MB


#### About
After dropping NaN values, we have lost **307** rows of data.

### Understanding Simple Metric for Numeric Columns

In [9]:
df_games.describe()

Year      NA_Sales      EU_Sales      JP_Sales   Other_Sales  \
count  16291.000000  16291.000000  16291.000000  16291.000000  16291.000000   
mean    2006.405561      0.265647      0.147731      0.078833      0.048426   
std        5.832412      0.822432      0.509303      0.311879      0.190083   
min     1980.000000      0.000000      0.000000      0.000000      0.000000   
25%     2003.000000      0.000000      0.000000      0.000000      0.000000   
50%     2007.000000      0.080000      0.020000      0.000000      0.010000   
75%     2010.000000      0.240000      0.110000      0.040000      0.040000   
max     2020.000000     41.490000     29.020000     10.220000     10.570000   

       Global_Sales  
count  16291.000000  
mean       0.540910  
std        1.567345  
min        0.010000  
25%        0.060000  
50%        0.170000  
75%        0.480000  
max       82.740000

As we can see, from Global Sales, we have a massive almost **50%** of the 16291 games selling less than 170k dollars worldwide.

## Final Objective

- **Predict Japanese Sales using North America, European and Other Sales, Genre and Platform.**

> This result is important for games that want to enter the Japenese Market, but don't know what to expect for their respective sales team. Also, it will be a good measure to invest (or not) in japanese translations.

### Columns to be used for Modelling

- Platform
- Genre
- NA_Sales
- EU_Sales
- JP_Sales
- Other_Sales

**obs.:**
- Global Sales is just the sum for all Sales columns.
- Platform and Genre need to have their respective key as a number for the purpose of modelling them into a machine learning algorithm.